bot/exts/christmas/adventofcode.py

```diff
@@ -131,28 +133,202 @@ async def day_countdown(bot: commands.Bot) -> None:
+        """Updates public leaderboards cached member amounts in every 30 minutes."""
+        # Check whether we are in the 25 days of advent
+        if not is_in_advent():
+            return
+
+        # Update every leaderboard with our session cookies
+        for aoc_id, cookie in self.leaderboard_cookies.items():
+            leaderboard = await AocPrivateLeaderboard.from_url(aoc_id, cookie)
+            await self.public_leaderboard_members.set(aoc_id, len(leaderboard.members))
+
+    async def refresh_leaderboard(self) -> None:
+        """Updates public PyDis leaderboard scores based on dates."""
+        self.last_updated = datetime.utcnow()
+        leaderboard_users = {}
+        leaderboards = [
+            await AocPrivateLeaderboard.json_from_url(aoc_id, cookie)
+            for aoc_id, cookie in self.leaderboard_cookies.items()
+        ]
+
+        # Check if any requests failed
+        if False in leaderboards:
+            log.warning("Unable to get one or more of the public leaderboards. Not updating cache.")
+            return
+
+        for leaderboard in leaderboards:
+            for member_id, data in leaderboard["members"].items():
+                leaderboard_users[int(member_id)] = {
+                    "name": data.get("name", "Anonymous User"),
+                    "aoc_id": int(member_id),
+                    "days": {
+                        day: {
+                            "star_one": "1" in stars,
+                            "star_two": "2" in stars,
+                            "star_one_earned": int(stars["1"]["get_star_ts"]) if "1" in stars else None,
+                            "star_two_earned": int(stars["2"]["get_star_ts"]) if "2" in stars else None,
+                        } for day, stars in data.get("completion_day_level", {}).items()
+                    }
+                }
+
+        # Iterate over every advent day
+        for day in range(1, 26):
+            day = str(day)
+            star_one_users = []
+            star_two_users = []
+
+            for user, user_data in leaderboard_users.items():
+                if day in user_data["days"]:
+                    if user_data["days"][day]["star_one"]:
+                        star_one_users.append({
+                            "id": user,
+                            "earned": datetime.fromtimestamp(user_data["days"][day]["star_one_earned"]),
+                        })
+
+                    if user_data["days"][day]["star_two"]:
+                        star_two_users.append({
+                            "id": user,
+                            "earned": datetime.fromtimestamp(user_data["days"][day]["star_two_earned"]),
+                        })
+
+            # Sort these lists based on the time a user earnt a star
+            star_one_users = sorted(star_one_users, key=lambda k: k["earned"])[:100]
+            star_two_users = sorted(star_two_users, key=lambda k: k["earned"])[:100]
```

Comment:
I know the original specification for this task was to give points, decrementing from 100, but I did more research and found this on private leaderboards:
> For N users, the first user to get each star gets N points, the second gets N-1, and the last gets 1.

This ultimately will not have a major effect on the final results, but if there is time, remove the list splicing on these two lines, and the `points = 100` should just become the length of leaderboard_users.

I won't request this as a change, as ultimately it's not necessary.